### ISSUE TITLE
Rule Change: Report invalid signed numeric radix literals in radix

### ISSUE THREAD

**Issue by Pixel998** [eslint/eslint#21012]
### What rule do you want to change?

radix

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/* eslint radix: "error" */

parseInt("10", -1);
parseInt("10", +37);
parseInt("10", -0);
```

### What does the rule currently do for this code?

The rule currently does not report any errors for these calls.

### What will the rule do after it's changed?

The rule will report these calls as invalid radix values, because the effective radix values are not integers between 2 and 36.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by mdjermanovic** [comment]
Makes sense to me. I think we can treat this as a semver-minor bug fix in this rule.
